prefetch/evaluate.cc:152:13: error: comparison is always false due to limited range of data type

I'm noticing the following compilation error on my m1 mac running the CI fedora:38 docker image:

```
╰─➤  uname -a
Linux docker-desktop 5.15.49-linuxkit-pr #1 SMP PREEMPT Thu May 25 07:27:39 UTC 2023 aarch64 GNU/Linux

╰─➤  make V=1
make[1]: Entering directory '/home/bneradt/src/ts_asf_master_cmake_the_experimental_plugins/plugins'
depbase=`echo prefetch/evaluate.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ../libtool  --tag=CXX   --mode=compile c++ -DHAVE_CONFIG_H -I. -I../include  -Dlinux -DDEBUG -D_DEBUG -D_LARGEFILE64_SOURCE=1 -D_COMPILE64BIT_SOURCE=1 -D_REENTRANT -D__STDC_LIMIT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -I/home/bneradt/src/ts_asf_master_cmake_the_experimental_plugins/include -I/home/bneradt/src/ts_asf_master_cmake_the_experimental_plugins/lib -I/home/bneradt/src/ts_asf_master_cmake_the_experimental_plugins/lib/swoc/include -I/home/bneradt/src/ts_asf_master_cmake_the_experimental_plugins/lib/yamlcpp/include -I/home/bneradt/src/ts_asf_master_cmake_the_experimental_plugins/lib/yamlcpp/include  -I/home/bneradt/src/ts_asf_master_cmake_the_experimental_plugins/lib/yamlcpp/include   -D_GNU_SOURCE -I/opt/openssl-quic/include -DOPENSSL_NO_SSL_INTERN -DOPENSSL_API_COMPAT=10002 -DOPENSSL_IS_OPENSSL3 -std=c++17 -ggdb3 -pipe -Wall -Wextra -Wno-ignored-qualifiers -Wno-unused-parameter -Wno-format-truncation -Wno-cast-function-type -Wno-stringop-overflow -Werror -Wno-invalid-offsetof -Wno-noexcept-type -Wsuggest-override  -MT prefetch/evaluate.lo -MD -MP -MF $depbase.Tpo -c -o prefetch/evaluate.lo prefetch/evaluate.cc &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  c++ -DHAVE_CONFIG_H -I. -I../include -Dlinux -DDEBUG -D_DEBUG -D_LARGEFILE64_SOURCE=1 -D_COMPILE64BIT_SOURCE=1 -D_REENTRANT -D__STDC_LIMIT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -I/home/bneradt/src/ts_asf_master_cmake_the_experimental_plugins/include -I/home/bneradt/src/ts_asf_master_cmake_the_experimental_plugins/lib -I/home/bneradt/src/ts_asf_master_cmake_the_experimental_plugins/lib/swoc/include -I/home/bneradt/src/ts_asf_master_cmake_the_experimental_plugins/lib/yamlcpp/include -I/home/bneradt/src/ts_asf_master_cmake_the_experimental_plugins/lib/yamlcpp/include -I/home/bneradt/src/ts_asf_master_cmake_the_experimental_plugins/lib/yamlcpp/include -D_GNU_SOURCE -I/opt/openssl-quic/include -DOPENSSL_NO_SSL_INTERN -DOPENSSL_API_COMPAT=10002 -DOPENSSL_IS_OPENSSL3 -std=c++17 -ggdb3 -pipe -Wall -Wextra -Wno-ignored-qualifiers -Wno-unused-parameter -Wno-format-truncation -Wno-cast-function-type -Wno-stringop-overflow -Werror -Wno-invalid-offsetof -Wno-noexcept-type -Wsuggest-override -MT prefetch/evaluate.lo -MD -MP -MF prefetch/.deps/evaluate.Tpo -c prefetch/evaluate.cc  -fPIC -DPIC -o prefetch/.libs/evaluate.o
prefetch/evaluate.cc: In function 'String {anonymous}::sub(StringView, StringView)':
prefetch/evaluate.cc:136:13: error: comparison is always false due to limited range of data type [-Werror=type-limits]
  136 |     if (val < 0) {
      |         ~~~~^~~
prefetch/evaluate.cc:152:13: error: comparison is always false due to limited range of data type [-Werror=type-limits]
  152 |     if (val < 0) {
      |         ~~~~^~~
cc1plus: all warnings being treated as errors
make[1]: *** [Makefile:6573: prefetch/evaluate.lo] Error 1
make[1]: Leaving directory '/home/bneradt/src/ts_asf_master_cmake_the_experimental_plugins/plugins'
make: *** [Makefile:7555: all-recursive] Error 1

╰─➤  c++ --version
c++ (GCC) 13.1.1 20230614 (Red Hat 13.1.1-4)
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

This is likely introduced via #9985.

--- a/plugins/prefetch/evaluate.cc
+++ b/plugins/prefetch/evaluate.cc
@@ -32,14 +32,14 @@
 namespace
 {
 
-inline char
+inline int8_t
 tonum(char ch)
 {
   return ch - '0';
 }
 
 inline char
-tochar(char ch)
+tochar(int8_t ch)
 {
   return ch + '0';
 }
@@ -74,7 +74,7 @@ add(StringView const lhs, StringView const rhs)
   auto ito = other.crbegin();
 
   while (ito != other.crend()) {
-    char val = tonum(*itr) + tonum(*ito);
+    int8_t val = tonum(*itr) + tonum(*ito);
     if (carry) {
       ++val;
     }
@@ -92,7 +92,7 @@ add(StringView const lhs, StringView const rhs)
   }
 
   while (result.rend() != itr && carry) {
-    char val = tonum(*itr) + 1;
+    int8_t val = tonum(*itr) + 1;
     if (val < 10) {
       carry = false;
     } else {
@@ -129,7 +129,7 @@ sub(StringView const lhs, StringView const rhs)
   auto itb = rhs.crbegin();
 
   while (result.rend() != itr && rhs.crend() != itb) {
-    char val = tonum(*itr) - tonum(*itb);
+    int8_t val = tonum(*itr) - tonum(*itb);
     if (borrow) {
       --val;
     }
@@ -148,7 +148,7 @@ sub(StringView const lhs, StringView const rhs)
 
   // keep pushing borrow
   while (result.rend() != itr && borrow) {
-    char val = tonum(*itr) - 1;
+    int8_t val = tonum(*itr) - 1;
     if (val < 0) {
       borrow  = true;
       val    += 10;

